go/runtime/txpool/main_queue.go

```diff
@@ -0,0 +1,158 @@
+package txpool
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/hash"
+	"github.com/oasisprotocol/oasis-core/go/runtime/host/protocol"
+)
+
+var (
+	_ UsableTransactionSource        = (*mainQueue)(nil)
+	_ RecheckableTransactionStore    = (*mainQueue)(nil)
+	_ RepublishableTransactionSource = (*mainQueue)(nil)
+)
+
+// MainQueueTransaction is a transaction and its metadata in the main queue.
+type MainQueueTransaction struct {
+	TxQueueMeta
+
+	// priority defines the transaction's priority as specified by the runtime.
+	priority uint64
+
+	// sender is a unique transaction sender identifier as specified by the runtime.
+	sender string
+	// senderSeq is a per-sender sequence number as specified by the runtime.
+	senderSeq uint64
+}
+
+func newTransaction(tx TxQueueMeta) *MainQueueTransaction {
+	return &MainQueueTransaction{
+		TxQueueMeta: tx,
+	}
+}
+
+// String returns a string representation of a transaction.
+func (tx *MainQueueTransaction) String() string {
+	return fmt.Sprintf("MainQueueTransaction{hash: %s, first_seen: %s, priority: %d}", tx.TxQueueMeta.Hash, tx.TxQueueMeta.FirstSeen, tx.priority)
+}
+
+// Raw returns the raw transaction data.
+func (tx *MainQueueTransaction) Raw() []byte {
+	return tx.TxQueueMeta.Raw()
```

Comment:
Given that `TxQueueMeta` is embedded this is forwarded by default. Same with some below.

Reply:
thanks, forwarding

Reply:
transferred doc comments too